src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs

```diff
@@ -1601,40 +1601,46 @@ private static bool IsAzureDevOpsArtifactsUrl(string location)
-
                 // Ensure the parent target directory has been created.
                 using (FileStream outStream = new FileStream(temporaryFileName,
-                                                      _options.Overwrite ? FileMode.Create : FileMode.CreateNew,
-                                                      FileAccess.Write))
+                                                        _options.Overwrite ? FileMode.Create : FileMode.CreateNew,
+                                                        FileAccess.Write))
                 {
-                    using (var response = await client.SendAsync(requestMessage))
-                    {
-                        response.EnsureSuccessStatusCode();
-
-                        using (var inStream = await response.Content.ReadAsStreamAsync())
+                    await ExponentialRetry.Default.RetryAsync(
+                        async () =>
                         {
-                            downloadOutput.AppendLine($"    {sourceUri} =>");
-                            foreach (string targetFile in targetFiles)
+                            if (authHeader != null)
                             {
-                                downloadOutput.AppendLine($"      {targetFile}");
+                                requestMessage = new HttpRequestMessage(HttpMethod.Get, sourceUri)
+                                {
+                                    Headers =
+                                    {
+                                        { HttpRequestHeader.Authorization.ToString(), authHeader.ToString() }
+                                    }
+                                };
                             }
-                            await inStream.CopyToAsync(outStream, cancellationToken);
-                        }
-                    }
+                            else
+                            {
+                                requestMessage = new HttpRequestMessage(HttpMethod.Get, sourceUri);
+                            }
+
+                            using (var response = await client.SendAsync(requestMessage))
+                            {
+                                response.EnsureSuccessStatusCode();
+
+                                using (var inStream = await response.Content.ReadAsStreamAsync())
+                                {
+                                    downloadOutput.AppendLine($"    {sourceUri} =>");
+                                    foreach (string targetFile in targetFiles)
+                                    {
+                                        downloadOutput.AppendLine($"      {targetFile}");
+                                    }
+                                    await inStream.CopyToAsync(outStream, cancellationToken);
+                                }
+                           }
+                        },
+                        ex => Console.WriteLine($"    Failed to download {sourceUri}: {ex.Message}"),
```

Comment:
This doesn't seem accurate... Since we are going to retry, telling the user we "failed" could be confusing.  It should probably contain the phrase "retrying" or something in it, so they at least know it's not an actual problem. (Or we could just say nothing, since they don't really need to know)

Reply:
At least, I assume that's what this callback is for?  I'm assuming the implementation of ExponentialRetry here. :-)

Reply:
I'll re-word it to make it clear that we are retrying.